xova/apps/xova/averaging.py

```diff
@@ -142,19 +162,35 @@ def average_main(main_ds, time_bin_secs, chan_bin_size,
     avg
         tuple containing averaged data
     """
-    # Get the appropriate spectral window and polarisation Dataset
-    # Must have a single DDID table
-
     output_ds = []
 
     for ds in main_ds:
-        if respect_flag_row is False:
-            ds = ds.assign(FLAG_ROW=(("row",), ds.FLAG.data.all(axis=(1, 2))))
+        compress = False
+
+        if fields and ds.FIELD_ID not in fields:
+            compress = True
+
+        if scan_numbers and ds.SCAN_NUMBER not in scan_numbers:
+            compress = True
```

Comment:
@bennahugo The if statements above can probably be changed to continue in order to simply ignore FIELDS and SCANs

Reply:
@sjperkins I've changed this to continue in the commit. We could have a switch to flip between this. However, generally speaking unselecting scans and fields should remove them from the product.

As we discussed yesterday this is properly better handled through a taql query, but for the moment the current selection is the minimum needed for production use.